### DISCUSSION TITLE
how do i do a blank screen and middle scroll in the modchart

### DISCUSSION THREAD

**Discussion by Assestion** [KadeArchive/Kade-Engine#978] [Q&A]
i know how to do effects and stuff like make notes jump and make bottom disappear but my concern is middle scroll and a blank screen like in fnf dark sheep mod, i have no idea how to do it, i cant find how in the docs, can someone please lend me a modchart of it, i need it for a mod

**Comment by puyoxyz** [comment]
middle scroll is in the modchart examples

**Reply by Real-Toonz** [reply]
not anymore lol

**Comment by NK-523** [comment]
This code is taken from the modchart of defeat, from VS Imposter. may kill you, I don't know.

`function setDefault(id)
	_G['defaultStrum'..id..'X'] = getActorX(id)
end


function start(song) -- arguments, the song name
    showOnlyStrums = true -- remove all hud elements besides notes and strums
    for i=0,3 do -- fade out the first 4 receptors (the ai receptors)
		tweenPosXAngle(i, _G['defaultStrum'..i..'X'] - 1000,getActorAngle(i) + 1000, 0.6, 'setDefault')
    end
    for i = 4, 7 do -- go to the center
        tweenPosXAngle(i, _G['defaultStrum'..i..'X'] - 275,getActorAngle(i) + 360, 0.6, 'setDefault')
    end
end

-- this gets called every frame
function update(elapsed) -- arguments, how long it took to complete a frame

end

-- this gets called every beat
function beatHit(beat) -- arguments, the current beat of the song
    
end

-- this gets called every step
function stepHit(step) -- arguments, the current step of the song (4 steps are in a beat)

end`

**Reply by NK-523** [reply]
removes misses, accuracy, and health counters also makes it middle scroll.

**Reply by HatD3V** [reply]
what type of file does it need to be?

**Reply by glisteningtwilights** [reply]
> what type of file does it need to be?

It needs to be called modchart
and the file extension needs to be .lua

**Comment by Wire-Firey** [comment]
Modchart for Middle Scroll (but no black screen):

function start(song) -- arguments, the song name
    showOnlyStrums = true
    for i=0,3 do
    tweenPosXAngle(i, _G['defaultStrum'..i..'X'] - 1000,getActorAngle(i) + 1000, 0.6, 'setDefault')
    end
    for i = 4, 7 do -- go to the center
    tweenPosXAngle(i, _G['defaultStrum'..i..'X'] - 275,getActorAngle(i) + 360, 0.6, 'setDefault')
    end
end
